src/parser.rs

```diff
@@ -25,6 +25,9 @@ pub trait Parser<'a, T: Input<'a>> {
     /// Reset a `Parser`.
     fn reset(&mut self);
 
+    /// Returns the queue of all matched `(Token, value)`s
+    fn queue_with_captures(&self) -> Vec<(Self::Token, String)>;
```

Comment:
As I said in an outdates comment, this might actually be useful outside of a debugging context so it might be useful to call it `queue_with_captures`?